### DISCUSSION TITLE
How to set "Runtime" Ruby 3.2 instead of default Python 3.10

### DISCUSSION THREAD

**Discussion by ajitAccenture** [GoogleCloudPlatform/cloud-foundation-fabric#2070] [Q&A]
**Describe the bug**
I am creating cloud function which is used to deploy ruby code from repo , but not able to find which varaible or enviroment varaible need to set Runtime" Ruby 3.2 instead of default Python 3.10.

**Environment**
```
module "cf_pubsub_trigger" {
  source      = "./fabric/modules/cloud-function-v2"
  project_id      = "modules-4071"
  name            = "moduletest-cf-pubsub-trigger"
  service_account = "SA-1@modules-95de4071.iam.gserviceaccount.com"
  bucket_name     = "bkt-cftest-00"
  bundle_config = {
    source_dir  =  "test-pubsub/src/"
    output_path = "test-archive.zip"
  }
  environment_variables = {GOOGLE_FUNCTION_SOURCE = "app.rb",  GOOGLE_ENTRY_POINT = "ruby app.rb", GOOGLE_RUNTIME_VERSION=3.2 }
  trigger_config = {
    event_type   = "google.cloud.pubsub.topic.v1.messagePublished"
    pubsub_topic = module.pubsub-topic.id
  }
  depends_on = [module.pubsub-topic]
  vpc_connector = {
    create          = false
    name            = "projectsdev-4743af0/locations/europe-west3/connectors/vpc-conn-risk-1"
    egress_settings = "ALL_TRAFFIC"
  }
}
```

```
output from `git rev-parse --short HEAD`
```

**To Reproduce**
Apply Above code

**Expected behavior**
It should show Runtime" Ruby 3.2 instead of default Python 3.10, so that cloud function deployed successfully with ruby code

**Result**
Geting below error
Error: Error waiting to create function: Error waiting for Creating function: Error code 3, message: Could not create or update Cloud Run service test-cf-pubsub-trigger, Container Healthcheck failed. Revision 'moduletest-cf-pubsub-trigger-00001-sub' is not ready and cannot serve traffic. The user-provided container failed to start and listen on the port defined provided by the PORT=8080 environment variable. Logs for this revision might contain more information.

<img width="755" alt="image" src="https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/assets/50900806/dec20f70-0e9f-48ff-9c48-94fce5a473c9">

**Additional context**
Add any other context about the problem here

**Comment by juliocc** [comment]
You're looking for [`function_config.runtime`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/cloud-function-v2/variables.tf#L72)

Please don't open issues for Q&A. Use this forum for that.
